numba/core/ssa.py

```diff
@@ -267,7 +267,12 @@ def on_assign(self, states, assign):
             if len(defmap) == 0:
                 newtarget = assign.target
                 _logger.debug("first assign: %s", newtarget)
-                assert newtarget.name in scope.localvars
+                if newtarget.name not in scope.localvars:
+                    wmsg = (
+                        f"variable name {newtarget.name!r} not in scope "
+                        f"in {assign} on line {assign.loc}",
+                    )
+                    warnings.warn(errors.NumbaIRAssumptionWarning(wmsg))
```

Comment:
```suggestion
                if newtarget.name not in scope.localvars:
                    wmsg = f"variable name {newtarget.name!r} not in scope"
                    warnings.warn(errors.NumbaIRAssumptionWarning(wmsg, loc=assign.loc))
```
Does it help to use Numba's error message `loc` handling here?

Reply:
oh, i forgot that it has that

Reply:
changed in https://github.com/numba/numba/pull/7223/commits/ec01acdbf837795b410bfce0cb238b5b3233a207